solc/CommandLineInterface.cpp

```diff
@@ -843,7 +849,11 @@ void CommandLineInterface::handleCombinedJSON()
 			output[g_strSources][sourceCode.first]["AST"] = converter.toJson(m_compiler->ast(sourceCode.first));
 		}
 	}
-	cout << dev::jsonCompactPrint(output) << endl;
+
+	if (m_args.count(g_argOutputDir))
+		createJson(m_args[g_argInputFile].as<vector<string>>().front(), output);
```

Comment:
Why did you decide to name the file after one of the input files? The combined json should contain the compilation result from all input files. What about `result.json` a filename?